delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/AclsRoutes.scala

```diff
@@ -0,0 +1,277 @@
+                      completeIO(status, acls.replace(address, aclEntity.toAcl, rev).map(_.void))
+                    }
+                  },
+                  (patch & entity(as[PatchAcl])) { aclPatch =>
+                    authorizeFor(address, aclsWrite).apply {
+                      aclPatch match {
+                        case AppendAcl(acl)   =>
+                          completeIO(acls.append(address, acl, rev).map(_.void))
+                        case SubtractAcl(acl) =>
+                          completeIO(acls.subtract(address, acl, rev).map(_.void))
+                      }
+                    }
+                  },
+                  delete {
+                    authorizeFor(address, aclsWrite).apply {
+                      completeIO(OK, acls.delete(address, rev).map(_.void))
+                    }
+                  },
+                  (get & parameter("self" ? true)) {
+                    case true  =>
+                      (parameter("rev".as[Long].?) & parameter("ancestors" ? false)) {
+                        case (Some(_), true)    => reject(simultaneousRevAndAncestorsRejection)
+                        case (Some(rev), false) =>
+                          completeIO(
+                            acls
+                              .fetchSelfAt(address, rev)
+                              .map { acl =>
+                                val searchResults = Seq(acl).flatten
+                                SearchResults(searchResults.size.toLong, searchResults)
+                                  .asInstanceOf[SearchResults[AclResource]]
+                              }
+                              .leftWiden[AclRejection]
+                          )
+                        case (None, true)       =>
+                          completeUnscoredSearch(
+                            acls
+                              .fetchSelfWithAncestors(address)
+                              .map(_.value.values.toSeq)
+                          )
+                        case (None, false)      =>
+                          completeUnscoredSearch(
+                            acls
+                              .fetchSelf(address)
+                              .map(Seq(_).flatten)
+                          )
+                      }
+                    case false =>
+                      authorizeFor(address, aclsRead).apply {
+                        (parameter("rev".as[Long].?) & parameter("ancestors" ? false)) {
+                          case (Some(_), true)    => reject(simultaneousRevAndAncestorsRejection)
+                          case (Some(rev), false) =>
+                            completeIO(
+                              acls
+                                .fetchAt(address, rev)
+                                .map { acl =>
+                                  val searchResults = Seq(acl).flatten
+                                  SearchResults(searchResults.size.toLong, searchResults)
```

Comment:
why asInstanceOf is needed? if the type is not recognized, fixing it in the `.map[SearchResults[AclResource]]` should be enough. Otherwise something has to be changed